tests/simulation/test_download.py

```diff
@@ -154,3 +159,44 @@ def test_match_filetypes():
     ]
 
     assert Downloader.match_filetypes(manifest) == expected_filetypes
+
+
+def test_compare_checksums():
+    parent_dir = pathlib.Path(__file__).parent.parent.absolute()
+    file_path = str(parent_dir / "mock_data" / "calib000000.txt")
+    checksum_path1 = str(parent_dir / "mock_data" / "checksum_calib000000.txt")
+    checksum_path2 = str(
+        parent_dir / "mock_data" / "wrong_checksum_calib000000.txt"
+    )
+
+    with not_raises(ChecksumError):
+        pytest.raises(compare_checksums(file_path, checksum_path1))
+    pytest.raises(ChecksumError, compare_checksums, file_path, checksum_path2)
+
+
+@contextmanager
+def not_raises(exception):
+    try:
+        yield
+    except exception:
+        raise pytest.fail("DID RAISE {0}".format(exception))
+
+
+@patch("datasetinsights.data.simulation.download.download_file_from_url")
+@patch("datasetinsights.data.simulation.download.compare_checksums")
+def test_run(mocked_download, mocked_checksum):
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--verbose", action="store_true", default=False)
+    parser.add_argument("--data-root", type=str, default="")
+    parser.add_argument(
+        "--name",
+        type=str,
+        default="GroceriesReal",
+        required=True,
+    )
+    parser.add_argument("--version", type=str, default="v3")
+
+    args = parser.parse_args()
```

Comment:
Why do you need to use `ArgumentParser` here? Can you directly mock up an `args` class here? 
If in the future, we added more public dataset to this method, does that mean we have to completely changer this test? Is there a better way to make it open for extension?

Reply:
Use `CN()` to mock args.